api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db.hibernate;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Field;
+import org.openmrs.FormField;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class HibernateFormDAOTest extends BaseContextSensitiveTest {
+	
+	private static final String FORM_XML = "org/openmrs/api/db/hibernate/include/HibernateFormDAOTestDataSet.xml";
+	
+	@Autowired
+	private HibernateFormDAO dao;
+	
+	@Before
+	public void setUp() {
+		executeDataSet(FORM_XML);
+	}
+		
+	@Test
+	public void shouldFilterAgainstFormFields() {
+		List<FormField> formFields = Arrays.asList(new FormField(2), new FormField(3));
+		Assert.assertEquals(1, (Object)dao.getForms(null, false, Collections.emptyList(), null, formFields, formFields, Collections.emptyList()).size());
+		
+		formFields = Arrays.asList(new FormField(2), new FormField(3), new FormField(5));
+		Assert.assertEquals(0, (Object)dao.getForms(null, false, Collections.emptyList(), null, formFields, formFields, Arrays.asList(new Field(3))).size());
```

Comment:
Doesn't the above unit test make enough coverage to the above change @dkayiwa ?

Reply:
No it does not, because this test still passes even when i revert your change from f.formId to form.formId

Reply:
Are you sure the tests don't @dkayiwa !

Try it locally boss! The tests pretty fail, I have just tried it :)

Reply:
Are you referring to the line where i put my comment?

Reply:
@samuelmale ping

Reply:
With that I'm curious to know what's really happening and how hibernate criterias work behind the box.